### PR TITLE
Add option to switch the Sequence register

### PR DESCRIPTION
**Pull request by HGSilveri**
This has been requested more than once, so I figured we could add a proper way to do it.

### CODE REVIEW

File: tests/test_sequence.py [hunk 1]
```diff
@@ -533,6 +533,64 @@ def test_ising_mode(
...
+    pulse = Pulse.ConstantPulse(1000, 1, -1, 2)
+    seq = init_seq(
+        reg,
+        DigitalAnalogDevice,
+        "raman",
+        "raman_local",
+        [pulse],
+        initial_target="q0",
+        parametrized=parametrized,
+    )
```
Comment (a-corni) [inline tests/test_sequence.py:548]: Do I get it right that you are not testing the conversion of a Sequence with a Mappable Register into a Sequence with a Register/another Mappable Register ? Is it worth testing ? Should we check what happens when DMM are configured in the Sequence as well ?

File: pulser-core/pulser/sequence/sequence.py [hunk 2]
```diff
@@ -684,6 +684,24 @@ def _config_detuning_map(
...
+            new_register: The new register to give the sequence.
+
+        Returns:
+            The sequence with the new register.
+        """
+        new_seq = type(self)(register=new_register, device=self._device)
+        # Copy the variables to the new sequence
+        new_seq._variables = self.declared_variables
+        for call in self._calls[1:] + self._to_build_calls:
+            getattr(new_seq, call.name)(*call.args, **call.kwargs)
```
Comment (a-corni) [inline pulser-core/pulser/sequence/sequence.py:702]: So the only condition on the new Register/Mappable Register is that the labels in the register match the labels used in local operations ? Perhaps we should explain this in the docstring ?

File: tests/test_sequence.py (hunk 1 above)
Comment (HGSilveri) [reply tests/test_sequence.py:548]: You are correct. It doesn't hurt to check these cases, so I'll just add them

File: pulser-core/pulser/sequence/sequence.py (hunk 2 above)
Comment (HGSilveri) [reply pulser-core/pulser/sequence/sequence.py]: I put some more info in the docstring, please take a look :)

File: pulser-core/pulser/sequence/sequence.py [hunk 3]
```diff
@@ -684,6 +684,44 @@ def _config_detuning_map(
...
+        This means that operations referecing specific qubits IDs
+        (eg. `Sequence.target()`) expect to find the same qubit IDs in the new
+        register. By the same token, switching from a register to a mappable
+        register might fail if one of the instructions does not work with
+        mappable registers (e.g. `Sequence.configure_slm_mask()`).
+
+        Warns:
+            UserWarning: If the sequence is configuring a detuning map, a
+            warning is raised to remind the user that the detuning map is
+            unchanged and might no longer be aligend with the new register.
```
Comment (a-corni) [inline pulser-core/pulser/sequence/sequence.py:703]: Perfect ! I was expecting an issue of this genre :)

**Review by a-corni — APPROVED** [review]
Lgtm ! Thanks for clarifying the docstring and adding the tests, it's way clearer now :)

**Pull request merged**